src/sql/src/plan/query.rs

```diff
@@ -763,9 +763,20 @@ pub fn eval_as_of<'a>(
     let evaled = ex.eval(&[], temp_storage)?;
 
     Ok(match ex.typ(desc.typ()).scalar_type {
-        ScalarType::Numeric { .. } => numeric::cx_datum()
-            .try_into_i128(evaled.unwrap_numeric().0)?
-            .try_into()?,
+        ScalarType::Numeric { .. } => {
+            let mut cx = numeric::cx_datum();
+            let mut d = evaled.unwrap_numeric().0;
+            if d.exponent() < 0 {
+                bail!("AF OF expressions do not support fractional milliseconds");
```

Comment:
```suggestion
                bail!("AS OF expressions do not support fractional milliseconds");
```

Reply:
🤦 will fix on the next round of edits; ty for catching this